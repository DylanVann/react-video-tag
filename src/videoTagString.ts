export interface VideoTagProps {
    class?: string
    className?: string
    src?: string
    poster?: string
    type?: string
    muted?: boolean
    autoPlay?: boolean
    playsInline?: boolean
    loop?: boolean
}

export const videoTagString = (props: VideoTagProps = {}): string => {
    const classOrClassName = props.className || props.class
    const classString = classOrClassName ? ` class="${classOrClassName}"` : ``
    const muted = props.muted ? ` muted` : ``
    const autoPlay = props.autoPlay ? ` autoplay` : ``
    const playsInline = props.playsInline ? ` playsinline` : ``
    const loop = props.loop ? ` loop` : ``
    const poster = props.poster ? ` poster` : ``
    const src = props.src ? ` src="${props.src}"` : ` src=""` // required
    const type = props.type ? ` type="${props.type}"` : ``

    return `<video${classString}${muted}${autoPlay}${playsInline}${loop}${poster}><source${src}${type}></video>`
}
