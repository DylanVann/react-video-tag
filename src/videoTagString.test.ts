import { videoTagString } from './videoTagString'

it('outputs a string', () => {
    expect(videoTagString()).toMatchSnapshot()
})

it('outputs a string with props', () => {
    expect(
        videoTagString({
            class: 'test-class',
            src: 'image.png',
            poster: 'image.png',
            type: 'mp4',
        }),
    ).toMatchSnapshot()
})
