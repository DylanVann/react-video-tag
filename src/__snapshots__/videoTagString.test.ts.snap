// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs a string 1`] = `"<video><source src=\\"\\"></video>"`;

exports[`outputs a string with props 1`] = `"<video class=\\"test-class\\" poster><source src=\\"image.png\\" type=\\"mp4\\"></video>"`;
